ee/hogai/graph/root/nodes.py

```diff
@@ -166,10 +132,11 @@ def _get_model(self, state: AssistantState, config: RunnableConfig):
         if settings.INKEEP_API_KEY:
             available_tools.append(search_documentation)
         for tool_name in self._get_contextual_tools(config).keys():
-            if tool_name not in CONTEXTUAL_TOOL_NAME_TO_TOOL_MODEL:
-                continue  # Possibly a deployment mismatch
-            available_tools.append(CONTEXTUAL_TOOL_NAME_TO_TOOL_MODEL[cast(AssistantContextualTool, tool_name)])
-
+            try:
```

Comment:
The tool abstraction here looks good. 

Apart from this RE: subgraph, I'm primarily looking for the root node to be swappable. I think this root node would still have access to search docs/create insight which SQL writing wouldn't need. So if I had sql editing as the contextual tool here, it should be the only tool that's callable (or it can ask just answer the user if it's a question that's not directly related to SQL creation).

Also, most of the root node prompt wouldn't be super necessary for just SQL writing